src/PSR7/Validators/SecurityValidator.php

```diff
@@ -165,6 +167,8 @@ private function validateApiKeySecurityScheme(
         RequestInterface $request,
         SecurityScheme $securityScheme
     ): void {
+        Assert::isInstanceOf($request, ServerRequestInterface::class);
```

Comment:
actually this should be guarded here. so all other signatures can just be fixed to be `ServerRequestInterface` directly, not via asserts

https://github.com/thephpleague/openapi-psr7-validator/pull/106/files#diff-be02d5fa32966a6c559f82d49d961f2236badc1858ca4f7c6ebc09b58387dbabR50-R52

Reply:
Are you suggesting to change the signature of `validateApiKeySecurityScheme` method?

Reply:
exactly, it's private and guarded - why not?

Reply:
Well, it is guarded to be `RequestInterface`, not `ServerRequestInterface` [here.](https://github.com/thephpleague/openapi-psr7-validator/blob/master/src/PSR7/Validators/SecurityValidator.php#L48) So we would have to change it also there to `ServerRequestInterface` Do you want that?

Reply:
I think it's better then adding more asserts which complicates reading

Reply:
Changed.